### PR TITLE
chore: make demo readme more clear for new developers

### PR DESCRIPTION
**Pull request by rthomare**
# Pull Request Checklist

- [X ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ X] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ X] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ X] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ X] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ X] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ X] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR updates the `README.md` in the `examples/ui-demo` directory to improve clarity and organization regarding setup instructions.

### Detailed summary
- Changed the section title from `## Getting Started` to `# Getting Started`.
- Added a new section `## Setup your Environment` with detailed steps for environment configuration.
- Introduced a new section `## Install Dependencies` with installation instructions.
- Added a new section `## Build the aa-sdk Libraries` with build instructions.
- Added a new section `## Start the Server` with instructions to start the server.
- Removed outdated commands for starting the server and replaced them with a streamlined command list.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by RobChangCA — DISMISSED** [review]

File: examples/ui-demo/README.md [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
+*In the root `/aa-sdk/` directory*
+
+```bash
+npm install
+# or
+yarn install
+# or
+pnpm install
+# or
+bun install
```
Comment (moldy530) [inline examples/ui-demo/README.md:20]: we use `yarn` everywhere so we should just have this be that

File: examples/ui-demo/README.md [hunk 2]
```diff
@@ -1,8 +1,40 @@
...
+*In the root `/aa-sdk/` directory*
+
+```bash
+npm build
+# or
+yarn build
+# or
+pnpm build
+# or
+bun build
```
Comment (moldy530) [inline examples/ui-demo/README.md:33]: same here

File: examples/ui-demo/README.md [hunk 3]
```diff
@@ -1,8 +1,40 @@
 This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
 
-## Getting Started
+# Getting Started
 
-First, run the development server:
+## Setup your Enviornment
```
Comment (dphilipson) [inline examples/ui-demo/README.md:5]: Nit: typo "enviornment" -> "environment", Also "your" should be capitalized in titles, it's only articles and prepositions that aren't.

File: examples/ui-demo/README.md [hunk 4]
```diff
@@ -1,8 +1,40 @@
...
+1. Ensure you have the right npm version corresponding to the `/aa-sdk/.nvmrc` file.
```
Comment (dphilipson) [inline examples/ui-demo/README.md:6]: That's actually the node version, not the npm version.

File: examples/ui-demo/README.md [hunk 5]
```diff
@@ -1,8 +1,40 @@
...
+2. Go to Alchemy's Dashboard, and setup an [account config](https://dashboard.alchemy.com/accounts/config/create).
+   1. For local developement set the configuration with a redirect url to http://localhost:3000.
```
Comment (dphilipson) [inline examples/ui-demo/README.md:8]: Typo: "developement" -> "development"

File: examples/ui-demo/README.md [hunk 6]
```diff
@@ -1,8 +1,40 @@
 This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
 
-## Getting Started
+# Getting Started
 
-First, run the development server:
+## Setup your Enviornment
+1. Ensure you have the right npm version corresponding to the `/aa-sdk/.nvmrc` file.
+2. Go to Alchemy's Dashboard, and setup an [account config](https://dashboard.alchemy.com/accounts/config/create).
```
Comment (dphilipson) [inline examples/ui-demo/README.md:7]: (Very nit) "setup" -> "set up".  "Setup" is a noun, "set up" is a verb.

**Review by dphilipson — APPROVED** [review]
I have some very nitpicky comments. I'm contributing!

**Review by dphilipson — APPROVED** [review]

**Pull request merged**